packages/berry-pnp/sources/loader/makeApi.ts

```diff
@@ -288,18 +288,15 @@ export function makeApi(runtimeState: RuntimeState, opts: MakeApiOptions): PnpAp
    * Forward the resolution to the next resolver (usually the native one)
    */
 
-  function callNativeResolution(request: string, issuer: string): string | false {
-    if (issuer.endsWith('/'))
-      issuer += 'internal.js';
-
-    request = NodeFS.fromPortablePath(request);
-    issuer = NodeFS.fromPortablePath(issuer);
+  function callNativeResolution(request: PortablePath, issuer: PortablePath): NativePath | false {
+    if (issuer.endsWith(`/`))
+      issuer = `${issuer}internal.js` as PortablePath;
 
     // Since we would need to create a fake module anyway (to call _resolveLookupPath that
     // would give us the paths to give to _resolveFilename), we can as well not use
     // the {paths} option at all, since it internally makes _resolveFilename create another
     // fake module anyway.
-    return Module._resolveFilename(request, makeFakeModule(issuer), false, {plugnplay: false});
+    return Module._resolveFilename(request, makeFakeModule(NodeFS.fromPortablePath(issuer)), false, {plugnplay: false});
```

Comment:
`makeFakeModule` could be typed too. I don't remember if it should be a Native or Portable Path...